src/tools/immutable_object_cache/ObjectCacheStore.cc

```diff
@@ -302,5 +357,71 @@ std::string ObjectCacheStore::get_cache_file_path(std::string cache_file_name,
   return m_cache_root_dir + cache_file_dir + cache_file_name;
 }
 
+void ObjectCacheStore::handle_throttle_ready(void* ctx, uint64_t flag) {
+  m_io_throttled.store(false);
+}
+
+bool ObjectCacheStore::take_token_from_throttle(uint64_t object_size,
+                                                uint64_t object_num) {
+  bool got_token = true;
+
+  if (m_io_throttled.load() == true) {
+    return true;
+  }
+
+  if (got_token && (m_qos_enabled_flag & ROC_QOS_IOPS_THROTTLE)) {
+    got_token =
+      m_throttles[ROC_QOS_IOPS_THROTTLE]->get(object_num, this, &ObjectCacheStore::handle_throttle_ready, nullptr, 0);
+  }
+
+  if (got_token && (m_qos_enabled_flag & ROC_QOS_BPS_THROTTLE)) {
```

Comment:
Here if blocked by IOPS,  got_token will be false, and stop promote, BPS throttle doesn't need to be done, even if it's on